service_contracts.rst

```diff
@@ -63,7 +63,148 @@ The PFS will be paid with signed IOUs, roughly a simplified uRaiden adapted to 1
 OneToN
 ======
 
-Specification in progress. See https://github.com/raiden-network/raiden-services/issues/38.
+Overview
+--------
+
+The OneToN contract handles payments for the PFS. It has been chosen with the
+following properties in mind:
+
+-  easy to implement
+-  low initial gas cost even when fees are paid to many PFSs
+-  a certain risk of double spends is accepted
+
+The concept is based on the idea to use a user's single deposit in the
+UDC as a security deposit for off-chain payments to all PFSs. The client
+sends an IOU consisting of (sender, receiver, amount, expiration,
+signature) to the PFS with every path finding request. The PFS verifies
+the IOU and checks that ``amount >= prev_amount + pfs_fee``. At any
+time, the PFS can claim the payment by submitting the IOU on-chain.
+Afterwards, no further IOU with the same (sender, receiver, expiration)
+can be claimed.
+
+Related:
+
+-  `https://github.com/raiden-network/team/issues/257`_
+-  `https://github.com/raiden-network/team/issues/256`_
+-  `https://gist.github.com/heikoheiko/214dbbd954e0f97e0e13b2fefdc7c753`_
+
+.. _`https://github.com/raiden-network/team/issues/257`: https://github.com/raiden-network/team/issues/257
+.. _`https://github.com/raiden-network/team/issues/256`: https://github.com/raiden-network/team/issues/256
+.. _`https://gist.github.com/heikoheiko/214dbbd954e0f97e0e13b2fefdc7c753`: https://gist.github.com/heikoheiko/214dbbd954e0f97e0e13b2fefdc7c753
+
+Requirements
+------------
+
+-  low latency (<1s)
+-  reliability, high probability of success (P > 0.99)
+-  low cost overhead (<5% of transferred amount)
+-  low fraud rate (< 3%, i.e. some fraud is tolerable)
+-  can be implemented quickly
+
+Communication between client and PFS
+------------------------------------
+
+When requesting a route, the IOU is added as five separate arguments to
+the `existing HTTP query params`_.
```

Comment:
I though it would go in the request header as it's done in uRaiden!? Does this bring advantages?

Reply:
And in this case GET is not a good idea, since we expect different results and different payment info each time.

Reply:
Ok, so we need to look into that anyways.